### PR TITLE
[PW-2895] Age and gender segments

### PR DESCRIPTION
**Pull request by toydestroyer**
# Refer to the issue id if any. Include the link to the issue. E.g:
[We need 2 more filter, age and gender](https://perxtechnologies.atlassian.net/browse/PW-2895)

# Describe the changes made. What does this PR changes that might be critical. If any critical decisions have been made, make sure you explain the rationale for these decisions.
- Age segments
- Gender segments

# How does the implementation addresses the problem

After merging this, we wiil be able to segment audience by gender and age.

### CODE REVIEW

**Comment by toydestroyer** [comment]
Reference: https://perxtechnologies.atlassian.net/wiki/spaces/WD/pages/685473793/Cognito+Service

File: services/cognito/app/operations/segments/age.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    private
+
+    def init(ctx, segment:, **)
+      ctx[:base_date] = Time.zone.today
+      ctx[:segment] = Array.wrap(segment)
+      ctx[:translated_segments] = []
+    end
+
+    def verify_segment(_ctx, segment:, **)
+      segment.each do |element|
```
Comment (rpbaltazar) [inline services/cognito/app/operations/segments/age.rb]: can't we just do `segment.all? {|s| verify_segment_element(s)}`

File: services/cognito/app/operations/segments/age.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    def incorrect_segment(_ctx, errors:, segment:, **)
+      errors.add(:segment, "Incorrect age segment value: #{segment}")
+    end
+
+    def translate_segment(_ctx, segment:, base_date:, translated_segments:, **)
+      segment.each do |element|
+        case element
+        when Hash
+          el = element.with_indifferent_access
+          translated_segments << ((base_date - el['to'].to_i.years - 1.year + 1.day)..(base_date - el['from'].to_i.years))
```
Comment (rpbaltazar) [inline services/cognito/app/operations/segments/age.rb]: I think this is worth to move to a date range class that is able to build these. it looks very convoluted code

File: services/cognito/app/operations/segments/age.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    def apply_segment(ctx, users:, translated_segments:, **)
+      ctx[:model] = users.where(birthday: translated_segments)
+    end
+
+    def segment_not_applied(_ctx, errors:, segment:, **)
+      errors.add(:model, "Can't apply age segment: #{segment}")
+    end
+
+    def verify_segment_element(element)
```
Comment (rpbaltazar) [inline services/cognito/app/operations/segments/age.rb]: verify_segment_element does not really translate to true or false. can we have a more clear naming?

**Pull request merged**